var/spack/repos/builtin/packages/vasp/package.py

```diff
@@ -90,7 +132,7 @@ def edit(self, spec, prefix):
         filter_file('^FFTW[ ]{0,}?=.*$', 'FFTW ?=', 'makefile.include')
         filter_file('^MPI_INC[ ]{0,}=.*$', 'MPI_INC ?=', 'makefile.include')
         filter_file('-DscaLAPACK.*$\n', '', 'makefile.include')
-        filter_file('^SCALAPACK*$', '', 'makefile.include')
+        filter_file('^SCALAPACK[ ]{0,}=.*$', '', 'makefile.include')
```

Comment:
The replace part was the important one since it is the part making use of the environment variable set at `set_build_environment`

```suggestion
        filter_file('^SCALAPACK[ ]{0,}=.*$', 'SCALAPACK ?=', 'makefile.include')
```

Reply:
Thanks @nrichart ,  I missed it somehow.